src/lib.rs

```diff
@@ -84,10 +90,15 @@ impl Server {
     ///     });
     /// }
     /// ```
-    pub fn new(
-        handler: fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>,
+    pub fn new<
+        H: Fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>
```

Comment:
In the definition of the `Handler` type could you omit the `Box` and use that here? IMHO it reads better this way.

Reply:
type aliases don't work with traits, so you can't.

Reply:
Ah, good point.

Reply:
heh, we went through all of that as we paired on this today. good thought! but yeah, alas.